File: Exercism Codes/C#/Series.cs
public static class Series
{
    public static string[] Slices(string numbers, int sliceLength)
    {
        if (sliceLength > numbers.Length || sliceLength <= 0 || numbers.Length == 0)
        {
            throw new ArgumentException();
        }

        List<string> result = [];
        for (int i = 0; i <= numbers.Length - sliceLength; i++)
        {
            result.Add(numbers.Substring(i, sliceLength));
        }
        return [..result];
    }
}

File: Exercism Codes/PHP/Series.php
<?php

declare(strict_types=1);

function slices(string $digits, int $series): array
{
    if ($series === 0 || $series > strlen($digits)) {
        throw new Exception();
    }

    $result = [];
    for ($i = 0; $i <= strlen($digits) - $series; $i++) {
        $result[] = substr($digits, $i, $series);
    }
    return $result;
}
